cosmwasm/contracts/crosschain-registry/src/helpers.rs

```diff
@@ -1,15 +1,208 @@
+use cosmwasm_std::{Addr, Deps};
 use osmosis_std_derive::CosmwasmExt;
 use sha2::{Digest, Sha256};
 
+use crate::execute::{FullOperation, Permission};
+use crate::state::{CHAIN_ADMIN_MAP, CHAIN_MAINTAINER_MAP, CONFIG, GLOBAL_ADMIN_MAP};
+use crate::ContractError;
+
+// check_is_contract_governor is used for functions that can only be called by the contract governor
+pub fn check_is_contract_governor(deps: Deps, sender: Addr) -> Result<(), ContractError> {
+    let config = CONFIG.load(deps.storage).unwrap();
+    if config.owner != sender {
+        Err(ContractError::Unauthorized {})
+    } else {
+        Ok(())
+    }
+}
+
+// check_permission checks if an account with the provided permission is authorized to perform the action requested
+// GlobalAdmin can add addresses to any address map, ChainAdmin can only add addresses to their
+// own chain map, and ChainMaintainer cant add addresses to any address map
+pub fn check_permission(
+    provided_permission: Permission,
+    max_permission: Permission,
```

Comment:
I was confused as to what `max_permission` meant here. Now I get that it's basically the role of the current user

Reply:
Clarified it here! https://github.com/osmosis-labs/osmosis/pull/4553/commits/93e2243184648db0c388043ea4e8ed97eec29aa8